src/main/java/net/dries007/tfc/network/PacketStackFood.java

```diff
@@ -0,0 +1,129 @@
+
+package net.dries007.tfc.network;
+
+import io.netty.buffer.ByteBuf;
+import net.dries007.tfc.TerraFirmaCraft;
+import net.dries007.tfc.api.capability.food.CapabilityFood;
+import net.dries007.tfc.api.capability.food.IFood;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
+import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+public class PacketStackFood implements IMessage
+{
+    private int slotNumber;
+
+    public PacketStackFood()
+    {
+    }
+
+    public PacketStackFood(int slotNumber)
+    {
+        this.slotNumber = slotNumber;
+    }
+
+    public static final class Handler implements IMessageHandler<PacketStackFood, IMessage>
+    {
+        @Override
+        public IMessage onMessage(PacketStackFood message, MessageContext ctx)
+        {
+            TerraFirmaCraft.getProxy().getThreadListener(ctx).addScheduledTask(() -> {
+                EntityPlayer player = TerraFirmaCraft.getProxy().getPlayer(ctx);
+                if (player != null)
+                {
+                    if (message.slotNumber < 0 || message.slotNumber >= player.openContainer.inventorySlots.size())
+                    {
+                        return;
+                    }
+
+                    Slot targetSlot = player.openContainer.getSlot(message.slotNumber);
+                    ItemStack targetStack = targetSlot.getStack();
+                    IFood targetCap = targetStack.getCapability(CapabilityFood.CAPABILITY, null);
+
+                    if (targetCap == null || targetStack.getMaxStackSize() == targetStack.getCount() || targetCap.isRotten())
+                    {
+                        return;
+                    }
+
+                    List<Slot> stackableSlots = getStackableSlots(targetSlot, player.openContainer.inventorySlots);
+                    int currentAmount = targetStack.getCount();
+                    int remaining = targetStack.getMaxStackSize() - currentAmount;
+                    long minCreationDate = targetCap.getCreationDate();
+
+                    Iterator<Slot> slotIterator = stackableSlots.iterator();
+                    while (remaining > 0 && slotIterator.hasNext())
+                    {
+                        Slot slot = slotIterator.next();
+                        ItemStack stack = slot.getStack();
+                        IFood cap = stack.getCapability(CapabilityFood.CAPABILITY, null);
+
+                        if (cap == null || cap.isRotten()) continue;
+
+                        if (cap.getCreationDate() < minCreationDate)
+                        {
+                            minCreationDate = cap.getCreationDate();
+                        }
```

Comment:
This suffers from the exact same problem as #1086. It needs to do a comparison on traits. Either stack ignoring traits (and then properly collecting common traits and stripping unused ones as per the description of #1086. Or, only stack foods that have the same traits. Might be worth adding a utility method for both cases, `CapabilityFood#areItemsStackable` which does exactly the checks here:
 - Are the foods both not rotten
 - Are the traits exactly the same

Reply:
Rotten check I already have, but instacking logic. It's a good point to put that in capability, thanks.

Reply:
As of b67eab6a018d6a74b1903bca2b545b11b7309c46 there are two functions added to `CapabilityFood` that may be used for this purpose:
 - `CapabilityFood#mergeItemStacksIgnoreCreationDate`, and
 - `CapabilityFood#areStacksStackableExceptCreationDate`.
The latter does not check rotten, as it completely ignores their creation date.